src/components/atoms/Countdown/Countdown.js

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from 'react'
+import './Countdown.scss'
+
+const diff = (a, b) => {
+  const distance = a - b
+
+  return {
+    day: Math.floor(distance / (1000 * 60 * 60 * 24)),
+    hour: Math.floor((distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60)),
+    min: Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60)),
+    sec: Math.floor((distance % (1000 * 60)) / 1000),
+  }
+}
+
+const Countdown = ({
+  endTime = new Date().toISOString(),
+  className = '',
+  ...props
+}) => {
+  const [now, setDate] = useState(+new Date())
+  const difference = diff(+new Date(endTime), now)
+
+  useEffect(() => {
+    const interval = setInterval(() => setDate(+new Date()), 1000)
+    return () => clearInterval(interval)
+  }, [])
+
+  return (
+    <figure className={`${className} countdown`} {...props}>
+      {Object.entries(difference).reduce(
+        (acc, [key, comp]) => (
+          <>
+            {acc}
+            {acc && <span className="countdown__separator">:</span>}
+            <span className="countdown__item">
+              <span className="countdown__item-time">
+                {`${comp}`.padStart(2, '0')}
+              </span>
+              <small className="countdown__item-label">{key}</small>
+            </span>
+          </>
+        ),
+        '',
+      )}
```

Comment:
Seeing a `reduce` here was kinda unexpected to me. What'd you think about doing a `map` instead?

Reply:
@korokd What is your reasoning for map over reduce?

Reply:
I'm asking for the change because it took me some time to get what was going on here. Most (basically all of) JSX I've seen transforming an array into a collection of HTML used `map`.

On a more objective note, with a `map` you only have to care about the specific data point you're transforming, whereas with this `reduce` you have to do a form of `look behind`, which is odd to me and goes somewhat against the "as functional as possible" philosophy that seems to be prevalent in both React itself and its ecosystem.

To be clear, I'm not against `reduce` itself, but kinda against it **in the JSX**.

Reply:
@korokd I've updated the code to utilize `map` instead of `reduce`. Should be much more readable now.